private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -455,6 +455,65 @@ export class TemplateServiceClient {
     return this.storageProvider.updateTemplate(queryTemplate, newTemplate);
   }
 
+  /**
+   * @public 
+   * Batch update template state only, template must already exist. 
+   * Can only publish & unpublish batch template versions. 
+   */
+  public async batchUpdateTemplate(templateId: string, requests: TemplateStateRequest[], token?: string): Promise<JSONResponse<Number>> {
+    let authCheck = this._checkAuthenticated(token);
+    if (!authCheck.success) {
+      return authCheck;
+    }
+
+    let authId = this.authProvider.getAuthIDFromToken(token || this.authProvider.token);
+    let userResponse = await this._getUser(authId);
+
+    if (!userResponse.success) {
+      return { success: false, errorMessage: userResponse.errorMessage };
+    }
+
+    // Check if version already exists
+    let response = await this.getTemplates(token, templateId);
+    if (!response.success || !response.result || response.result.length === 0) {
+      return { success: false };
```

Comment:
you can add errorMessage here as well I guess, letting caller know why success is false

Reply:
Fixed, thanks!